libs/ssl-config/src/main/java/org/elasticsearch/common/ssl/SslConfiguration.java

```diff
@@ -0,0 +1,160 @@
+    private final List<String> supportedProtocols;
+
+    public SslConfiguration(SslTrustConfig trustConfig, SslKeyConfig keyConfig, SslVerificationMode verificationMode,
+                            SslClientAuthenticationMode clientAuth, List<String> ciphers, List<String> supportedProtocols) {
+        this.trustConfig = trustConfig;
+        this.keyConfig = keyConfig;
+        this.verificationMode = verificationMode;
+        this.clientAuth = clientAuth;
+        this.ciphers = Collections.unmodifiableList(ciphers);
+        this.supportedProtocols = Collections.unmodifiableList(supportedProtocols);
+    }
+
+    public SslTrustConfig getTrustConfig() {
+        return trustConfig;
+    }
+
+    public SslKeyConfig getKeyConfig() {
+        return keyConfig;
+    }
+
+    public SslVerificationMode getVerificationMode() {
+        return verificationMode;
+    }
+
+    public SslClientAuthenticationMode getClientAuth() {
+        return clientAuth;
+    }
+
+    public List<String> getCipherSuites() {
+        return ciphers;
+    }
+
+    public List<String> getSupportedProtocols() {
+        return supportedProtocols;
+    }
+
+    /**
+     * @return A collection of files that are used by this SSL configuration. If the contents of these files change, then any
+     * subsequent call to {@link #createSslContext()} (or similar methods) may create a context with different behaviour.
+     * It is recommended that these files be monitored for changes, and a new ssl-context is created whenever any of the files are modified.
+     */
+    public Collection<Path> getDependentFiles() {
+        Set<Path> paths = new HashSet<>(keyConfig.getDependentFiles());
+        paths.addAll(trustConfig.getDependentFiles());
+        return paths;
+    }
+
+    /**
+     * Dynamically create a new SSL context based on the current state of the configuration.
+     * Because the {@link #getKeyConfig() key config} and {@link #getTrustConfig() trust config} may change based on the
+     * contents of their referenced files (see {@link #getDependentFiles()}, consecutive calls to this method may
+     * return ssl-contexts with different configurations.
+     */
+    public SSLContext createSslContext() {
+        try {
+            final X509ExtendedKeyManager keyManager = keyConfig.createKeyManager();
+            final X509ExtendedTrustManager trustManager = trustConfig.createTrustManager();
+            SSLContext sslContext = SSLContext.getInstance(contextAlgorithm());
+            sslContext.init(new X509ExtendedKeyManager[] { keyManager }, new X509ExtendedTrustManager[] { trustManager }, null);
+            return sslContext;
+        } catch (RuntimeException e) {
+            throw e;
+        } catch (Exception e) {
+            throw new SslConfigException("cannot create ssl context", e);
+        }
+    }
+
+    /**
+     * Picks the best (highest security / most recent standard) SSL/TLS algorithm that is supported by the
```

Comment:
```suggestion
     * Picks the best (highest security / most recent standard) SSL/TLS version that is supported by the
```

Reply:
I resolved this with a slightly different wording. Let me know if you still see an issue.